nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -300,6 +300,14 @@ def setup(self, trainer: pl.Trainer) -> None:
         self.accelerator.setup(trainer)
         self.trainer = trainer
 
+        try:
+            self.model.optim.lr_scheduler.max_steps = trainer.max_steps
```

Comment:
Is this the right place to set this? Or does it need to happen before calling `trainer.fit`?

Reply:
Looks like this [line](https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/optim/base.py#L88) is triggered by `setup_megatron_parallel`->`setup_optimizers`, [which](https://github.com/NVIDIA/NeMo/blob/main/nemo/lightning/pytorch/strategies/megatron_strategy.py#L313) starts later in Strategy `setup()`. I checked with breakpoint to make sure scheduler's max_steps is correct when the scheduler setup runs. So I think this is early enough. I'll add an assertion at train start for this

Reply:
see 58c44363997f06544109ad78333c2b8825225cb7